gnome-image-installer/pages/install/gis-install-page.c

```diff
@@ -590,19 +665,46 @@ gis_install_page_shown (GisPage *page)
 {
   GisInstallPage *install = GIS_INSTALL_PAGE (page);
   GisInstallPagePrivate *priv = gis_install_page_get_instance_private (install);
-  gchar *msg = g_strdup_printf (_("Step %d of %d"), 1, 2);
+  GtkWidget *titlebar;
+  GList *children, *l;
+  gchar *msg;
+
+  msg = g_strdup_printf (_("Step %d of %d"), 1, 2);
   gtk_label_set_text (OBJ (GtkLabel*, "install_label"), msg);
   g_free (msg);
 
   gis_driver_save_data (GIS_PAGE (page)->driver);
 
   priv->client = udisks_client_new_sync (NULL, NULL);
 
-  if (gis_store_get_error() == NULL)
-    g_idle_add ((GSourceFunc)gis_install_page_verify, page);
-  else
-    gis_assistant_next_page (gis_driver_get_assistant (page->driver));
+  if (gis_store_get_error () != NULL)
+    {
+      gis_assistant_next_page (gis_driver_get_assistant (page->driver));
+      return;
+    }
+
+  g_idle_add ((GSourceFunc)gis_install_page_verify, page);
+
+  /* Hide the titlebar widgets */
+  titlebar = gis_assistant_get_titlebar (gis_driver_get_assistant (page->driver));
+  children = gtk_container_get_children (GTK_CONTAINER (titlebar));
 
+  gtk_header_bar_set_show_close_button (GTK_HEADER_BAR (titlebar), FALSE);
+
+  for (l = children; l != NULL ; l = l->next)
+    gtk_widget_hide (l->data);
```

Comment:
`update_navigation_buttons` in `gis-assistant.c` already has logic for hiding navigation buttons on the last page. Rather than duplicating it here, perhaps the page can have a vfunc for whether the buttons should be hidden – or a method to call to explicitly hide the buttons?